TestWebKitAPI test utilities are not declared in consistent header files

kkinnunen-apple opened the pull request:
#### 4ce7e49408a7a9cd5da160938b606a10f7aebe7e
<pre>
TestWebKitAPI test utilities are not declared in consistent header files
<a href="https://bugs.webkit.org/show_bug.cgi?id=246947">https://bugs.webkit.org/show_bug.cgi?id=246947</a>

Reviewed by Antti Koivisto.

Tests have following utility code needs:
 - Support for GTest to interact with custom types: printing types
   using them in expectations
 - Common code used in many different test files

Before, these were defined in inconsistent manner in many different
header files, sometimes with duplicate responsibilities.

After, the GTest support and utilities are in following files:

 - Test.h -- the main GTest support file. Includes GTest support for
   WTF types, as WTF is currently always used.

 - WTFTestUtilities.h -- Utilities for WTF tests.

 - WebCoreTestUtilities.h -- GTest support for WebCore types and
   utilities for tests testing WebCore code.

Move GTest utility macros to Test.h. This header is almost always needed
for tests anyway, it is the top-level test utility file.

Move WTF::String support for GTest into Test.h. This is almost always
needed for tests anyway, and all tests should include Test.h.

Move WTF related utilities to WTFTestUtilities.h

Move WebCore related utilities to WebCoreTestUtilities.h

Fix all tests that use Test.h as the primary header file to work
around the above bug. Instead, include Test.h as the include rules
mandate.

Rename couple of test files that do not contain Tests.cpp suffix.
Currently the suffix is used to trigger the style checker to know
that the file does not have a primary header file. This is a good
suffix anyway, as it removes linker errors on Cocoa, due to
object files having duplicate base name (CrossThreadCopier.cpp
in WTF and in TestWebKitAPI).

* Tools/TestWebKitAPI/CMakeLists.txt:
* Tools/TestWebKitAPI/DebugUtilities.h:
* Tools/TestWebKitAPI/Sources.txt:
* Tools/TestWebKitAPI/Test.h:
(WTF::operator&lt;&lt;):
* Tools/TestWebKitAPI/TestNotificationProvider.cpp:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/CustomProtocolsSyncXHRTest.mm:
* Tools/TestWebKitAPI/Tests/IPC/ConnectionTests.cpp:
* Tools/TestWebKitAPI/Tests/IPC/StreamConnectionBufferTests.cpp:
* Tools/TestWebKitAPI/Tests/IPC/StreamConnectionTests.cpp:
* Tools/TestWebKitAPI/Tests/JavaScriptCore/JSRunLoopTimer.mm:
* Tools/TestWebKitAPI/Tests/Misc/TestRunnerTests.cpp:
* Tools/TestWebKitAPI/Tests/WTF/CompletionHandlerTests.cpp:
* Tools/TestWebKitAPI/Tests/WTF/CrossThreadCopierTests.cpp: Renamed from Tools/TestWebKitAPI/Tests/WTF/CrossThreadCopier.cpp.
* Tools/TestWebKitAPI/Tests/WTF/HexNumber.cpp:
* Tools/TestWebKitAPI/Tests/WTF/JSONValue.cpp:
* Tools/TestWebKitAPI/Tests/WTF/RunLoop.cpp:
* Tools/TestWebKitAPI/Tests/WTF/StackTraceTest.cpp:
* Tools/TestWebKitAPI/Tests/WTF/StringBuilder.cpp:
* Tools/TestWebKitAPI/Tests/WTF/StringCommon.cpp:
* Tools/TestWebKitAPI/Tests/WTF/StringConcatenate.cpp:
* Tools/TestWebKitAPI/Tests/WTF/StringOperators.cpp:
* Tools/TestWebKitAPI/Tests/WTF/StringParsingBuffer.cpp:
* Tools/TestWebKitAPI/Tests/WTF/SuspendableWorkQueueTests.cpp: Renamed from Tools/TestWebKitAPI/Tests/WTF/SuspendableWorkQueue.cpp.
* Tools/TestWebKitAPI/Tests/WTF/URL.cpp:
* Tools/TestWebKitAPI/Tests/WTF/URLParser.cpp:
* Tools/TestWebKitAPI/Tests/WTF/WTFString.cpp:
* Tools/TestWebKitAPI/Tests/WTF/cocoa/URLExtras.mm:
* Tools/TestWebKitAPI/Tests/WebCore/ColorTests.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/DNS.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/FileMonitor.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/ImageBufferTests.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/Logging.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/PrivateClickMeasurement.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/PublicSuffix.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/PushMessageCrypto.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/SecurityOrigin.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/StringWithDirection.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/URLParserTextEncoding.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/cocoa/PrivateClickMeasurementCocoa.mm:
* Tools/TestWebKitAPI/Tests/WebCore/cocoa/TestGraphicsContextGLCocoa.mm:
* Tools/TestWebKitAPI/Tests/WebKit/CookieStorageFile.cpp:
* Tools/TestWebKitAPI/Tests/WebKit/DidRemoveFrameFromHiearchyInPageCache.cpp:
* Tools/TestWebKitAPI/Tests/WebKit/EnvironmentUtilitiesTest.cpp:
* Tools/TestWebKitAPI/Tests/WebKit/LoadCanceledNoServerRedirectCallback_Bundle.cpp:
* Tools/TestWebKitAPI/Tests/WebKit/UserMessage.cpp:
* Tools/TestWebKitAPI/Tests/WebKit/WKBackForwardListTests.mm: Renamed from Tools/TestWebKitAPI/Tests/WebKit/WKBackForwardList.mm.
* Tools/TestWebKitAPI/Tests/WebKit/WKPageConfiguration.cpp:
* Tools/TestWebKitAPI/Tests/WebKit/WillLoad.cpp:
* Tools/TestWebKitAPI/Tests/WebKit/curl/Certificates.cpp:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/DragAndDropTests.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ModalContainerObservation.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SOAuthorizationTests.mm: Renamed from Tools/TestWebKitAPI/Tests/WebKitCocoa/TestSOAuthorization.mm.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKAttachmentTests.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/_WKWebAuthenticationPanel.mm:
* Tools/TestWebKitAPI/Tests/WebKitLegacy/mac/AccessingPastedImage.mm:
* Tools/TestWebKitAPI/Tests/WebKitLegacy/mac/ClosingWebView.mm:
* Tools/TestWebKitAPI/Tests/WebKitLegacy/mac/DeallocWebViewInEventListener.mm:
* Tools/TestWebKitAPI/Tests/WebKitObjC/PreventImageLoadWithAutoResizing.mm:
* Tools/TestWebKitAPI/Tests/WebKitObjC/PreventImageLoadWithAutoResizing_Bundle.cpp:
* Tools/TestWebKitAPI/Tests/WebKitObjC/WKBrowsingContextGroupTest.mm:
* Tools/TestWebKitAPI/Tests/WebKitObjC/WKBrowsingContextLoadDelegateTest.mm:
* Tools/TestWebKitAPI/Tests/ios/ActionSheetTests.mm:
* Tools/TestWebKitAPI/Tests/ios/FocusPreservationTests.mm:
* Tools/TestWebKitAPI/Tests/ios/ShareSheetTests.mm:
* Tools/TestWebKitAPI/Tests/mac/AttributedString.mm:
* Tools/TestWebKitAPI/Tests/mac/ContextMenuCanCopyURL.mm:
* Tools/TestWebKitAPI/Tests/mac/ContextMenuTests.mm:
* Tools/TestWebKitAPI/Tests/mac/FontManagerTests.mm:
* Tools/TestWebKitAPI/Tests/mac/ImmediateActionTests.mm:
* Tools/TestWebKitAPI/Tests/mac/NSResponderTests.mm:
* Tools/TestWebKitAPI/Tests/mac/PageVisibilityStateWithWindowChanges.mm:
* Tools/TestWebKitAPI/Tests/mac/ScrollingCoordinatorTests.mm:
* Tools/TestWebKitAPI/Tests/mac/WebViewCanPasteURL.mm:
* Tools/TestWebKitAPI/Tests/mac/WebViewCanPasteZeroPng.mm:
* Tools/TestWebKitAPI/Tests/mac/WillPerformClientRedirectToURLCrash.mm:
* Tools/TestWebKitAPI/Tests/mac/WillSendSubmitEvent.mm:
* Tools/TestWebKitAPI/Utilities.h:
* Tools/TestWebKitAPI/WTFStringUtilities.cpp: Removed.
* Tools/TestWebKitAPI/WTFTestUtilities.h: Renamed from Tools/TestWebKitAPI/WTFStringUtilities.h.
(TestWebKitAPI::utf16String):
* Tools/TestWebKitAPI/WebCoreTestUtilities.cpp: Renamed from Tools/TestWebKitAPI/TestUtilities.cpp.
* Tools/TestWebKitAPI/WebCoreTestUtilities.h: Renamed from Tools/TestWebKitAPI/TestUtilities.h.
(TestWebKitAPI::ScopedSetAuxiliaryProcessTypeForTesting::ScopedSetAuxiliaryProcessTypeForTesting):
(TestWebKitAPI::ScopedSetAuxiliaryProcessTypeForTesting::~ScopedSetAuxiliaryProcessTypeForTesting):
* Tools/TestWebKitAPI/WebCoreUtilities.h: Removed.

Canonical link: <a href="https://commits.webkit.org/256398@main">https://commits.webkit.org/256398@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fb17934bfa6ad441ded53cf3367fc4afb71e3620

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/95422 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/4695 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/28472 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/105004 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/165270 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/99409 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/4713 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/33425 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/87788 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/100877 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/101086 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/3442 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/82035 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/30512 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/85334 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/87249 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/73352 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/86490 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/39211 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/18803 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/81789 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/36914 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/20099 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/28120 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4424 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/40893 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/42760 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/84467 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/42882 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/39344 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/19083 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/TestWebKitAPI/WTFTestUtilities.h:
```diff
@@ -30,33 +30,20 @@
+namespace TestWebKitAPI {
 
 template<size_t length>
-static String utf16String(const char16_t (&url)[length])
```
A bit weird to see a remove `static` declaration in a header file, aren't we going to have multiple definitions of this function now?

kkinnunen-apple replied:
It's a template, those are defined like that.
static template definition means each compilation unit that uses the function gets a compilation unit local template instantiation.

JonWBedard replied:
🤦‍♂️ Missed the `template<size_t length>` on the the line above.

JonWBedard commented on Tools/TestWebKitAPI/config.h:
```diff
@@ -130,3 +130,11 @@
+
+#if defined(BUILDING_TEST_WTF)
+// FIXME(https://bugs.webkit.org/show_bug.cgi?id=247121): Re-defining templates with different implementations is undefined behavior.
+#ifndef WTF_STRINGTYPEADAPTER_COPIED_WTF_STRING
```
I'm still unclear what's going on here, what are we using this macro for?

kkinnunen-apple replied:
It's hack-altering the implementation of the templates in StringConcatenation.h in order to count the copies. The copy counts are then asserted in StringOperations.cpp to check that the expected amount of copies is made.

kkinnunen-apple committed:
TestWebKitAPI test utilities are not declared in consistent header files
https://bugs.webkit.org/show_bug.cgi?id=246947

Reviewed by Antti Koivisto.

Tests have following utility code needs:
 - Support for GTest to interact with custom types: printing types
   using them in expectations
 - Common code used in many different test files

Before, these were defined in inconsistent manner in many different
header files, sometimes with duplicate responsibilities.

After, the GTest support and utilities are in following files:

 - Test.h -- the main GTest support file. Includes GTest support for
   WTF types, as WTF is currently always used.

 - WTFTestUtilities.h -- Utilities for WTF tests.

 - WebCoreTestUtilities.h -- GTest support for WebCore types and
   utilities for tests testing WebCore code.

Move GTest utility macros to Test.h. This header is almost always needed
for tests anyway, it is the top-level test utility file.

Move WTF::String support for GTest into Test.h. This is almost always
needed for tests anyway, and all tests should include Test.h.

Move WTF related utilities to WTFTestUtilities.h

Move WebCore related utilities to WebCoreTestUtilities.h

Fix all tests that use Test.h as the primary header file to work
around the above bug. Instead, include Test.h as the include rules
mandate.

Rename couple of test files that do not contain Tests.cpp suffix.
Currently the suffix is used to trigger the style checker to know
that the file does not have a primary header file. This is a good
suffix anyway, as it removes linker errors on Cocoa, due to
object files having duplicate base name (CrossThreadCopier.cpp
in WTF and in TestWebKitAPI).

* Tools/TestWebKitAPI/CMakeLists.txt:
* Tools/TestWebKitAPI/DebugUtilities.h:
* Tools/TestWebKitAPI/Sources.txt:
* Tools/TestWebKitAPI/Test.h:
(WTF::operator<<):
* Tools/TestWebKitAPI/TestNotificationProvider.cpp:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/CustomProtocolsSyncXHRTest.mm:
* Tools/TestWebKitAPI/Tests/IPC/ConnectionTests.cpp:
* Tools/TestWebKitAPI/Tests/IPC/StreamConnectionBufferTests.cpp:
* Tools/TestWebKitAPI/Tests/IPC/StreamConnectionTests.cpp:
* Tools/TestWebKitAPI/Tests/JavaScriptCore/JSRunLoopTimer.mm:
* Tools/TestWebKitAPI/Tests/Misc/TestRunnerTests.cpp:
* Tools/TestWebKitAPI/Tests/WTF/CompletionHandlerTests.cpp:
* Tools/TestWebKitAPI/Tests/WTF/CrossThreadCopierTests.cpp: Renamed from Tools/TestWebKitAPI/Tests/WTF/CrossThreadCopier.cpp.
* Tools/TestWebKitAPI/Tests/WTF/HexNumber.cpp:
* Tools/TestWebKitAPI/Tests/WTF/JSONValue.cpp:
* Tools/TestWebKitAPI/Tests/WTF/RunLoop.cpp:
* Tools/TestWebKitAPI/Tests/WTF/StackTraceTest.cpp:
* Tools/TestWebKitAPI/Tests/WTF/StringBuilder.cpp:
* Tools/TestWebKitAPI/Tests/WTF/StringCommon.cpp:
* Tools/TestWebKitAPI/Tests/WTF/StringConcatenate.cpp:
* Tools/TestWebKitAPI/Tests/WTF/StringOperators.cpp:
* Tools/TestWebKitAPI/Tests/WTF/StringParsingBuffer.cpp:
* Tools/TestWebKitAPI/Tests/WTF/SuspendableWorkQueueTests.cpp: Renamed from Tools/TestWebKitAPI/Tests/WTF/SuspendableWorkQueue.cpp.
* Tools/TestWebKitAPI/Tests/WTF/URL.cpp:
* Tools/TestWebKitAPI/Tests/WTF/URLParser.cpp:
* Tools/TestWebKitAPI/Tests/WTF/WTFString.cpp:
* Tools/TestWebKitAPI/Tests/WTF/cocoa/URLExtras.mm:
* Tools/TestWebKitAPI/Tests/WebCore/ColorTests.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/DNS.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/FileMonitor.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/ImageBufferTests.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/Logging.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/PrivateClickMeasurement.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/PublicSuffix.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/PushMessageCrypto.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/SecurityOrigin.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/StringWithDirection.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/URLParserTextEncoding.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/cocoa/PrivateClickMeasurementCocoa.mm:
* Tools/TestWebKitAPI/Tests/WebCore/cocoa/TestGraphicsContextGLCocoa.mm:
* Tools/TestWebKitAPI/Tests/WebKit/CookieStorageFile.cpp:
* Tools/TestWebKitAPI/Tests/WebKit/DidRemoveFrameFromHiearchyInPageCache.cpp:
* Tools/TestWebKitAPI/Tests/WebKit/EnvironmentUtilitiesTest.cpp:
* Tools/TestWebKitAPI/Tests/WebKit/LoadCanceledNoServerRedirectCallback_Bundle.cpp:
* Tools/TestWebKitAPI/Tests/WebKit/UserMessage.cpp:
* Tools/TestWebKitAPI/Tests/WebKit/WKBackForwardListTests.mm: Renamed from Tools/TestWebKitAPI/Tests/WebKit/WKBackForwardList.mm.
* Tools/TestWebKitAPI/Tests/WebKit/WKPageConfiguration.cpp:
* Tools/TestWebKitAPI/Tests/WebKit/WillLoad.cpp:
* Tools/TestWebKitAPI/Tests/WebKit/curl/Certificates.cpp:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/DragAndDropTests.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ModalContainerObservation.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SOAuthorizationTests.mm: Renamed from Tools/TestWebKitAPI/Tests/WebKitCocoa/TestSOAuthorization.mm.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKAttachmentTests.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/_WKWebAuthenticationPanel.mm:
* Tools/TestWebKitAPI/Tests/WebKitLegacy/mac/AccessingPastedImage.mm:
* Tools/TestWebKitAPI/Tests/WebKitLegacy/mac/ClosingWebView.mm:
* Tools/TestWebKitAPI/Tests/WebKitLegacy/mac/DeallocWebViewInEventListener.mm:
* Tools/TestWebKitAPI/Tests/WebKitObjC/PreventImageLoadWithAutoResizing.mm:
* Tools/TestWebKitAPI/Tests/WebKitObjC/PreventImageLoadWithAutoResizing_Bundle.cpp:
* Tools/TestWebKitAPI/Tests/WebKitObjC/WKBrowsingContextGroupTest.mm:
* Tools/TestWebKitAPI/Tests/WebKitObjC/WKBrowsingContextLoadDelegateTest.mm:
* Tools/TestWebKitAPI/Tests/ios/ActionSheetTests.mm:
* Tools/TestWebKitAPI/Tests/ios/FocusPreservationTests.mm:
* Tools/TestWebKitAPI/Tests/ios/ShareSheetTests.mm:
* Tools/TestWebKitAPI/Tests/mac/AttributedString.mm:
* Tools/TestWebKitAPI/Tests/mac/ContextMenuCanCopyURL.mm:
* Tools/TestWebKitAPI/Tests/mac/ContextMenuTests.mm:
* Tools/TestWebKitAPI/Tests/mac/FontManagerTests.mm:
* Tools/TestWebKitAPI/Tests/mac/ImmediateActionTests.mm:
* Tools/TestWebKitAPI/Tests/mac/NSResponderTests.mm:
* Tools/TestWebKitAPI/Tests/mac/PageVisibilityStateWithWindowChanges.mm:
* Tools/TestWebKitAPI/Tests/mac/ScrollingCoordinatorTests.mm:
* Tools/TestWebKitAPI/Tests/mac/WebViewCanPasteURL.mm:
* Tools/TestWebKitAPI/Tests/mac/WebViewCanPasteZeroPng.mm:
* Tools/TestWebKitAPI/Tests/mac/WillPerformClientRedirectToURLCrash.mm:
* Tools/TestWebKitAPI/Tests/mac/WillSendSubmitEvent.mm:
* Tools/TestWebKitAPI/Utilities.h:
* Tools/TestWebKitAPI/WTFStringUtilities.cpp: Removed.
* Tools/TestWebKitAPI/WTFTestUtilities.h: Renamed from Tools/TestWebKitAPI/WTFStringUtilities.h.
(TestWebKitAPI::utf16String):
* Tools/TestWebKitAPI/WebCoreTestUtilities.cpp: Renamed from Tools/TestWebKitAPI/TestUtilities.cpp.
* Tools/TestWebKitAPI/WebCoreTestUtilities.h: Renamed from Tools/TestWebKitAPI/TestUtilities.h.
(TestWebKitAPI::ScopedSetAuxiliaryProcessTypeForTesting::ScopedSetAuxiliaryProcessTypeForTesting):
(TestWebKitAPI::ScopedSetAuxiliaryProcessTypeForTesting::~ScopedSetAuxiliaryProcessTypeForTesting):
* Tools/TestWebKitAPI/WebCoreUtilities.h: Removed.

Canonical link: https://commits.webkit.org/256398@main
